python-frontend/src/main/java/org/sonar/python/semantic/v2/SymbolsModuleTypeProvider.java

```diff
@@ -69,6 +75,52 @@ public ModuleType createModuleType(List<String> moduleFqn, ModuleType parent) {
       .orElseGet(() -> createEmptyModule(moduleName, parent));
   }
 
+  PythonType resolveLazyTypeForFqn(String fullyQualifiedName) {
+    String[] split = fullyQualifiedName.split("\\.");
+    ModuleType currentModule = rootModule;
+    for (int i = 0; i < split.length; i++) {
+      String currentElement = split[i];
+      Optional<PythonType> pythonType = currentModule.resolveMember(currentElement);
+      if (pythonType.isEmpty()) {
+        // We should probably return an "UnresolvedType" here, keeping the FQN available for possible checks
+        return PythonType.UNKNOWN;
+      }
+      PythonType currentType = pythonType.get();
+      if (i == split.length - 1) {
+        // We have resolved the type we were looking for, return it
+        return currentType;
+      }
+      if (!(currentType instanceof ModuleType moduleType)) {
+        break;
+      }
+      // We have resolved an enclosing module, continue searching
+      currentModule = moduleType;
+    }
+    return PythonType.UNKNOWN;
+  }
+
+  public PythonType resolveLazyType(LazyType lazyType) {
+    if (lazyTypesContext.resolvedLazyTypes().containsKey(lazyType)) {
+      return lazyTypesContext.resolvedLazyTypes().get(lazyType);
```

Comment:
This line is uncovered, and I think because `FunctionType#returnType` now automatically resolves lazy types when called, I don't think it can be triggered anymore. Still, I prefer to keep this defensively, as it will be possible to trigger it when we resolve more types as lazy types.

Reply:
as discussed this is not needed anymore